### DISCUSSION TITLE
Why doesn't media library view look more like web version? Current media library view on android TV looks messy

### DISCUSSION THREAD

**Discussion by Daikamar** [jellyfin/jellyfin-androidtv#3390] [General]
I really want to switch from Plex to Jellyfin, but the Android TV version of Jellyfin just doesn't look/feel right. The web version is nice and I think what most people expect coming over from something like Plex, but for some reason the Android TV version went with a "minimalist" view or something that is just hard to use.
My main gripes are:
- Can't see TV/Movie titles until you select them at which point they are displayed at the top. I really just want titles under the posters so I can browse a page worth of shows without highlighting each. I have several shows in the library that auto-selected posters that don't have the show title on them and this little addition would go really far.
- Adding titles would also help with how cluttered the posters feel. They are too close to one another and while there is an option to increase/decrease poster size, there isn't any option for poster spacing.
- In addition to the general poster spacing, the background being replaced with a blown-up version of the poster for the currently selected show further exacerbates the issue as it blends the gaps between show posters and can cause what accounts to "blending" of posters that neighbor each other. An option to disable that feature would help, but ultimately, given how "filled" the screen is already, I don't see any value in replacing the background with the poster as it is already all but totally obscured.

Generally curious if I'm alone in these issues or not, but would love to hear feedback as well as any plans to address these CX concerns. I am an engineer and willing to contribute, but I presume CX changes along these lines would require approval of some sort (haven't gotten around to reading the contribution wikis yet).

**Comment by nielsvanvelzen** [comment]
The browsing UI in the app right now is not great (although it did improve somewhat overtime). The points you make are all fair. And you're not the only one that wants it improved (I've been harassed because that's how you get volunteers to fix your problems apparently 🤷). I have an [experimental branch](https://github.com/jellyfin/jellyfin-androidtv/compare/master...nielsvanvelzen:jellyfin-androidtv:compose-browser) to completely rewrite browsing that is in extremely early stages. But my main focus is on other parts of the app right now and the UI library we want to use (Jetpack Compose) is still not working properly on TV (focus issues).

![98500002c5](https://github.com/jellyfin/jellyfin-androidtv/assets/2305178/370ff444-1159-45db-b742-6fd37436d692)

**Reply by Daikamar** [reply]
Thanks for this follow-up. Nice to hear that this is something that's being looked into. Curious about your experimental branch and the focus issues you're seeing (anecdotally a build of your branch seems to work without problems in an emulator for me).

**Reply by nielsvanvelzen** [reply]
There is currently one focus issue that is a deal breaker, and that is the interoperability between "legacy" Android views and Compose views. When you switch focus from an Android view to compose it doesn't update. Since the experimental browser branch is fully compose you'll only notice it when you initially open the browser, it won't have an item focused by default because that is just not possible right now.

A different branch that I made over a year ago is the [compose-nowplayingview](https://github.com/nielsvanvelzen/jellyfin-androidtv/tree/compose-nowplayingview) branch. This rewrites the "now playing" button you see in the old browser when playing music. The focus issue is extremely obvious in this one.

Google is still working on TV support for Compose so issues like this one are to be expected. We're slowly adopting compose for non-interactive elements and when the focus issues are fixed we will migrate more and more.

**Reply by nylimited** [reply]
Just wanted to add a coupe of things as food for thought ...  the AndroidTV app is difficult to navigate, indeed. Not impossible, but difficult.

As an example, navigating sideways - TV shows as an example - is odd compared to other systems (JF web or Plex) and users are confused by it.

Placing buttons at the bottom of the display (i.e. TV Guide or genre or whatnot) is cumbersome and most renditions show these choices on top or near the top of the screen.

Incidentally, the loading of the TV Guide fails if one has too many channels.  Yes, I know this is not a proper bug report, sorry. Just mentioning it in case someone happens to be playing with that part of the code.

Sadly I am not a programmer to give a hand here but I do appreciate everyone's time and effort.   Would love to be more involved but I lack knowledge and skill set.  :(

**Comment by Blurayx** [comment]
Any news regarding Android TV looking more like the web version?

Thanks

**Reply by Zachwang007** [reply]
i also look forward to the NEW UI,just don't know when

**Reply by YaPaY** [reply]
> i also look forward to the NEW UI,just don't know when

me too

**Reply by nicgravel** [reply]
Looking for that too....want to move from Plex but this  is needed ...

**Reply by Blurayx** [reply]
I hope with the recent  beta releases, the UI gets a real makeover to look like the web version. A good start would be to have the A-Z  search on the right and thumb view be highlighted rather than enlarged when selecting. I have a large collection and it's very annoying having the A-Z search at the top. Please update the Android version.
Thanks

**Comment by Blurayx** [comment]
With the next Android TV update, can you please  make the media library view look more like web version?

Thanks

**Reply by nielsvanvelzen** [reply]
If someone paid me a full time salary for at least a month I can guarantee it yes.

**Reply by Blurayx** [reply]
Your answer was rather impolite.

**Reply by nielsvanvelzen** [reply]
You're right - my apologies for that. I get questions like this often and shouldn't have replied like this. I work on Jellyfin in my spare time and I'm well aware that the app could use improvements in the UI/UX department, and I have been working on that for years now. But I have high quality standards and am strict on keeping existing functionality working.
If I chose to create a brand new app from scratch all of this would've been much easier/quicker. But then a lot of niche functionality that only a small set of users rely on would be missing.
In addition, the framework I want to build the new user interface on wasn't usable on television apps for a long time, and only recently have we been able to use it in more places. As a result the recent 0.19 release already contains a subset of changes for the navigation on top and the search page.

I have work in progress versions of new settings, a new video player and most importantly the browsing and home screens. With a bit of luck the next version (0.20) will see a lot of these work in progress projects being finalized.

So to answer your question: I can't promise anything but I truly hope to finally make some big changes in the next update.

**Reply by Blurayx** [reply]
Thanks for responding. Apology accepted. All water under the bridge. 😀

I know you do all this in your spare time. Thanks for all your hard work you have done to maintain Jellyfin, and I truly respect what you do.    
Do you have a donation  link?

Thanks

**Reply by nielsvanvelzen** [reply]
You can find information for donating, including my links, on our website: https://jellyfin.org/docs/general/contributing/direct-donations

**Comment by ISbitPro** [comment]
the thing I miss most in the web app is simply LISTS
Don't know why the world of UI is moving away from lists, but I firmly believe that a list is the best way of listing things.
I get confused about the left to right thing. Specially since it hides previous episodes, so I cant easily scroll back

**Comment by Webrow** [comment]
@nielsvanvelzen We need to talk about that month salary. Jokes aside, do you have an issue or roadmap where I could try to contribute? Otherwise Ill happily throw in a donation if that gets some progress on UI end (I really dont care for it, but my dauhter likes to read the titles)

**Reply by nielsvanvelzen** [reply]
There is a roadmap for the Android apps: https://github.com/orgs/jellyfin/projects/47. However, I don't do a great job of updating it and I also don't really follow it. The things I work on are the things I'm motivated to work on, and sometimes that's the user interface and other times it is not.

I do have a few experiments for an improved browsing interface (that does show the titles!!) but it's far from ready, as that part of the app is extremely complicated because it needs to deal with all item types Jellyfin has (which we have almost 40 of!!) that come with all kinds of edge cases. For the upcoming 0.20 version there are already some changes that are needed for a new browsing experience though. Like the new design of the item cards.

Right now I don't expect much change to the browsing in the 0.20 update as I hope to release that soonish. I'll hopefully have some more time in a few weeks (it's currently very busy at my work) so I can dedicate some time on Jellyfin again.

**Comment by Webrow** [comment]
Understandable, I am not going to scaffold something into the current
master without any basis for compose there, ill wait for 0.20 and help from
there if I can.

**Comment by Webrow** [comment]
Respect for your guys project btw, lets not forget. And good luck on your
irl job
